Owlcat/PlotAntennas.py

```diff
@@ -0,0 +1,488 @@
+    This function is an implementation based on the following reference:
+
+    https://docs.hisparc.nl/coordinates/HiSPARC_coordinates.pdf
+
+    Parameters
+    ----------:
+    lat: :obj:`float`
+        Latitude in radians
+    lon: :obj:`float`
+        Longitude in radians
+    alt: :obj:`float`
+        Altitude in metres
+
+    Returns
+    -------
+    X, Y, Z:  ECEF coordinates in metres
+    """
+
+    logging.debug("Converting WGS84 to ECF")
+
+    # set up earth's shape ellipsoid approximation
+    # semi major axis
+    a = 6378137.0
+
+    # flattening
+    f = 1 / 298.257223563
+
+    # semi-minor axis
+    b = a - a * f
+
+    # eccentricity
+    e = np.sqrt((2 * f) - (f**2))
+
+    # Normal: Distance between a location on the ellipsoid and the
+    # intersection of its nromal and the ellipsoid's z-axis
+    N = a / np.sqrt(1 - e**2 * np.sin(lat)**2)
+
+    # altitude
+    h = alt
+
+    X = (N + h) * np.cos(lat) * np.cos(lon)
+    Y = (N + h) * np.cos(lat) * np.sin(lon)
+    Z = (((b**2 / a**2) * N) + h) * np.sin(lat)
+
+    return X, Y, Z
+
+
+def ecef_to_enu(x, y, z, r_lon, r_lat, r_alt):
+    """
+    Convert ECEF coordinates to East, North, Up (m). Because the ENU is a
+    local coordinate system (to the observer/ antenna in this case)
+    an reference position / origin is required from which calculate the
+    relative positions.
+
+    Parameters
+    ----------
+    x: :obj:`float`
+        X in metres
+    y: :obj:`float`
+        Y in metres
+    z: :obj:`float`
+        Z in metres
+    r_lon: :obj:`float`
+        Reference longitude in radians
+    r_lat: :obj:`float`
+        Reference latitude in radians
+    r_alt: :obj:`float`
+        Reference altitude in metres
+
+    Returns
+    -------
+    ee, en, eu: :obj:`float`
+        East, North Up positions in metres
+    """
+
+    logging.debug("Converting ECEF to ENU")
+
+    # convert reference coordinates to ECEF
+    rx, ry, rz = wgs84_to_ecef(r_lon, r_lat, r_alt)
+
+    dx = x - rx
+    dy = y - ry
+    dz = z - rz
+
+    ee = (-np.sin(r_lon) * dx) \
+        + (np.cos(r_lon) * dy) \
+        + (0 * dz)
+
+    en = (-np.sin(r_lat) * np.cos(r_lon) * dx) \
+        + (-np.sin(r_lat) * np.sin(r_lon) * dy) \
+        + (np.cos(r_lat) * dz)
+
+    eu = (np.cos(r_lat) * np.cos(r_lon) * dx) \
+        + (np.cos(r_lat) * np.sin(r_lon) * dy) \
+        + (np.sin(r_lat) * dz)
+
+    return ee, en, eu
+
+
+def wgs84_to_enu(lon, lat, alt, r_lon, r_lat, r_alt):
+    """Convert WSG84 / ITRF to ENU"""
+    X, Y, Z = wgs84_to_ecef(lon, lat, alt)
+    e, n, u = ecef_to_enu(X, Y, Z, r_lon, r_lat, r_alt)
+
+    return e, n, u
+
+
+def get_antenna_data(ms_name):
+    """Return antenna ITRF positions and other data"""
+
+    logging.debug("Getting antenna data")
+
+    try:
+        # get antenna related information
+        ant_sub = table("::".join((ms_name, "ANTENNA")), ack=False)
```

Comment:
You should use the ```with table(...) as ant_sub``` construct here to ensure the table is closed in event of an error. This ensures ```self.__exit__``` is called. Otherwise you need a finally clause here to ensure the objects are closed.

Reply:
Noted. Changing this